internal/controller/server.go

```diff
@@ -62,11 +67,14 @@ func (s *MyServer) actionError(w http.ResponseWriter, e string) {
 }
 
 func (s *MyServer) actionCreateURL(w http.ResponseWriter, r *http.Request) {
+	var currActionName contextKeyType = "actionName"
 	s.Logger.Debugln("Start ActionCreateUrl")
 
 	var answerStatus = http.StatusCreated
 	var userid int
 
+	ctx := context.WithValue(context.Background(), currActionName, "createurl")
```

Comment:
```suggestion
	ctx := context.WithValue(r.Context(), currActionName, "createurl")
```
Контекст необходимо брать не из воздуха, а из запроса. Причина очень простая - если пользователь прерывает запрос или отваливается соединение, то контекст запроса "протухает". Соответственно, выполняющийся в данном хендлере для данного пользователя SQL-запрос становится ненужным и рвется по протухшему контексту, снимая нагрузку с БД.

Reply:
поправил.